homeassistant/components/plugwise_stick/config_flow.py

```diff
@@ -0,0 +1,131 @@
+"""Config flow for the Plugwise_stick platform."""
+import os
+from typing import Dict
+
+import plugwise
+from plugwise.exceptions import NetworkDown, PortError, StickInitError, TimeoutException
+import serial.tools.list_ports
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant, callback
+
+from .const import CONF_USB_PATH, DOMAIN
+
+CONF_MANUAL_PATH = "Enter Manually"
+
+
+@callback
+def plugwise_stick_entries(hass: HomeAssistant):
+    """Return existing connections for Plugwise USB-stick domain."""
+    return {
+        (entry.data[CONF_USB_PATH])
+        for entry in hass.config_entries.async_entries(DOMAIN)
+    }
+
+
+class PlugwiseConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    async def async_step_user(self, user_input=None):
+        """Step when user initializes a integration."""
+        errors = {}
+        ports = await self.hass.async_add_executor_job(serial.tools.list_ports.comports)
+        list_of_ports = [
+            f"{p}, s/n: {p.serial_number or 'n/a'}"
+            + (f" - {p.manufacturer}" if p.manufacturer else "")
+            for p in ports
+        ]
+        list_of_ports.append(CONF_MANUAL_PATH)
+
+        if user_input is not None:
+            user_selection = user_input[CONF_USB_PATH]
+            if user_selection == CONF_MANUAL_PATH:
+                return await self.async_step_manual_path()
+            if user_selection in list_of_ports:
+                port = ports[list_of_ports.index(user_selection)]
+                device_path = await self.hass.async_add_executor_job(
+                    get_serial_by_id, port.device
+                )
+            else:
+                device_path = await self.hass.async_add_executor_job(
+                    get_serial_by_id, user_selection
+                )
+            errors = await validate_connection(self.hass, device_path)
+            if not errors:
+                return self.async_create_entry(
```

Comment:
You should add add this here:
```
                await self.async_set_unique_id("unique_id, raise_on_progress=False)
                self._abort_if_unique_id_configured()
```
The config_flow also needs a unique_id, otherwise you will be able to add the integration more than once. Also, there will be issues when adding a 2nd Stick. I know, unlikely, but there's always someone that will do this :)

Reply:
Thanks, I'll add an `unique_id` to the the config entry.